api/services/control/control.proto

```diff
@@ -25,8 +25,10 @@ service Control {
 }
 
 message PruneRequest {
-	repeated string filter = 1; // FIXME: not implemented
+	repeated string filter = 1;
 	bool all = 2;
+	int64 keepDuration = 3 [(gogoproto.nullable) = true];
```

Comment:
This could have been implemented as a filter string to keep the API more clean. containerd's filter package doesn't yet support the compares needed but it could be picked out and used until that package works.